### PR TITLE
Store assessments for rescoring

### PR DESCRIPTION
**Pull request by Jeff-Thompson12**
Addresses #536 

Stores the assessment tibble/metrics as bits so that the app can pull them back from the database to rescore without reassessing.

### CODE REVIEW

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK not really a new feature but thought you might want a look anyways.

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
Ahhhhh, the re-scoring process was SO much faster. Much needed! Thank you @Jeff-Thompson12! It's kind of a new feature, right? Could you please update the `NEWS` file and increment the dev version?

Also, I've been noticing this new "No encoding supplied" message that I think comes from the new version of `riskmetric`. Have you noticed it? I'm not sure what the warning is about regarding uploading the `samplesizeCMH` package, but it seemed to work just fine in the app.

image

**Comment by Jeff-Thompson12** [comment]
> Also, I've been noticing this new "No encoding supplied" message that I think comes from the new version of `riskmetric`. Have you noticed it? I'm not sure what the warning is about regarding uploading the `samplesizeCMH` package, but it seemed to work just fine in the app.
> 
> image

Yes, I have noticed and it is related to `riskmetric`. I have not looked into it beyond that though.

File: R/utils_insert_db.R [hunk 1]
```diff
@@ -16,12 +16,14 @@
 #' 
 #' @returns nothing
 #' @noRd
-dbUpdate <- function(command, db_name = golem::get_golem_options('assessment_db_name'), .envir = parent.frame()){
+dbUpdate <- function(command, db_name = golem::get_golem_options('assessment_db_name'), .envir = parent.frame(), params = NULL){
```
[the remark this answers is not in the document]
Comment (Jeff-Thompson12) [reply R/utils_insert_db.R:19]: > And not your fault, but maybe fix the "Environment" spelling error in `@param .envir Environemtn to evaluate each expression in`

That's where you are probably wrong. It's just not my fault with regards to this PR.

File: R/utils_insert_db.R [hunk 2]
```diff
@@ -200,8 +203,9 @@ insert_riskmetric_to_db <- function(pkg_name,
              riskmetric_assess[[metric$name]][[1]][1:length(riskmetric_assess[[metric$name]])]))
     
     dbUpdate(
-      "INSERT INTO package_metrics (package_id, metric_id, weight, value) 
-      VALUES ({package_id}, {metric$id}, {metric$weight}, {metric_value})", db_name
+      "INSERT INTO package_metrics (package_id, metric_id, value, encode) 
+      VALUES ({package_id}, {metric$id}, {metric_value}, $pkg_assess)", db_name,
+      params = list(pkg_assess = assessment_serialized[metric$name,])
```
[the remark this answers is not in the document]
Comment (Jeff-Thompson12) [reply R/utils_insert_db.R:209]: This was a fun PR. I have never used the BLOB data type in SQL before. At first I was going to trying using `dput()`, but I felt like we could run into weird behavior with storing and evaluating strings like that. Then I came across a blog post that used this method to store information related to a graph and realized that this should solve our issue with reproducibility which is why we reassessed in the first place.

[the remark this answers is not in the document]
Comment (AARON-CLARK) [reply R/utils_insert_db.R:209]: I've never seen the BLOB either! How neat is that?

**Review by AARON-CLARK — APPROVED** [review]
Off the ⛓️

**Pull request merged**